tests/v2/validation/token/token_test.go

```diff
@@ -0,0 +1,74 @@
+package token
+
+import (
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	fv3 "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/kubeapi/tokens"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"testing"
+)
+
+const (
+	initialTokenDesc = "my-token"
+	updatedTokenDesc = "changed-token"
+)
+
+type TokenTestSuite struct {
+	suite.Suite
+	client       *rancher.Client
+	session      *session.Session
+	cluster      *management.Cluster
+	createdToken *fv3.Token
+	patchedToken *v3.Token
+}
+
+func (t *TokenTestSuite) TearDownSuite() {
+	t.session.Cleanup()
+}
+
+func (t *TokenTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	t.session = testSession
+
+	client, err := rancher.NewClient("", t.session)
+	require.NoError(t.T(), err)
+
+	t.client = client
+}
+
+func (t *TokenTestSuite) TestPatchToken() {
+
+	subSession := t.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := t.client.WithSession(subSession)
+	require.NoError(t.T(), err)
+	t.client = client
+
+	tokenToCreate := &fv3.Token{Description: initialTokenDesc}
+	createdToken, err := t.client.Management.Token.Create(tokenToCreate)
+	require.NoError(t.T(), err)
+
+	t.createdToken = createdToken
+	assert.Equal(t.T(), initialTokenDesc, t.createdToken.Description)
+
+	patchedToken, unstructuredRes, err := tokens.PatchToken(t.client, t.client.RancherConfig.ClusterName, createdToken.Name, "replace", "/description", updatedTokenDesc)
+	require.NoError(t.T(), err)
+
+	t.patchedToken = patchedToken
+	assert.Equal(t.T(), updatedTokenDesc, t.patchedToken.Description)
+
+	uc := unstructuredRes.UnstructuredContent()
+	if val, ok := uc["groupPrincipals"]; ok {
+		assert.NotEmpty(t.T(), val)
```

Comment:
Curious, what are we expecting here for the value? can we also assert the expected value and the actual value are equal ?

Reply:
Where the fix is in Rancher I expect there not to be a value in the unstructured response for group principals. If you run this against 2.7.0 it will fail because the unstructured response has nil as a group principal value. If you run it against 2.7-head it shouldn't exist so this is handling both.